src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -262,6 +262,36 @@ private ConstantValue DefaultSyntaxValue
                         }
                     }
 
+                    // Ensure availability of `DecimalConstantAttribute`.
+                    // For lambdas, check also explicit attribute application
+                    // because it will be re-synthesized in the delegate type.
```

Comment:
This feels wrong. Why would we want to perform a check for a situation when there is no synthesized delegate?

Reply:
Makes sense, moved the check to `Binder.BindToInferredDelegateType`, i.e., where conversion to the synthesized delegate happens. That also allowed me to remove check for methods from `Conversions.GetMethodGroupDelegateConversion`.

Reply:
Actually, I would like to return to this approach. Why do you think this isn't a suitable place to do the check? If we have a lambda, it will always result in a synthesized delegate, right?